### PR TITLE
PYTHON-3777 add types to command_cursor.py

### PR DESCRIPTION
**Pull request by sleepyStick**
https://jira.mongodb.org/browse/PYTHON-3777

### CODE REVIEW

**Review by blink1073 — APPROVED** [review]
LGTM!

File: pymongo/command_cursor.py [hunk 1]
```diff
@@ -51,7 +71,7 @@ def __init__(
         self.__collection: Collection[_DocumentType] = collection
         self.__id = cursor_info["id"]
         self.__data = deque(cursor_info["firstBatch"])
-        self.__postbatchresumetoken = cursor_info.get("postBatchResumeToken")
+        self.__postbatchresumetoken: int = cast(int, cursor_info.get("postBatchResumeToken"))
```
Comment (ShaneHarvey) [inline pymongo/command_cursor.py:74]: postBatchResumeToken should be an optional mapping, not an int.

File: pymongo/command_cursor.py [hunk 2]
```diff
@@ -158,7 +178,7 @@ def _maybe_pin_connection(self, sock_info):
             else:
                 self.__sock_mgr = sock_mgr
 
-    def __send_message(self, operation):
+    def __send_message(self, operation: Union[_Query, _GetMore]) -> None:
```
Comment (ShaneHarvey) [inline pymongo/command_cursor.py:181]: operation can only be _GetMore here, not _Query.

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: pymongo/command_cursor.py (hunk 1 above)
Comment (sleepyStick) [reply pymongo/command_cursor.py]: Changed!

File: pymongo/command_cursor.py (hunk 2 above)
Comment (sleepyStick) [reply pymongo/command_cursor.py]: thanks for catching that!

**Review by ShaneHarvey — APPROVED** [review]

**Pull request merged**